src/wp-admin/media-upload.php

```diff
@@ -35,8 +35,8 @@
 // Require an ID for the edit screen.
 if ( isset( $action ) && 'edit' === $action && ! $ID ) { // phpcs:ignore WordPress.NamingConventions.ValidVariableName
 	wp_die(
-		'<h1>' . __( 'Something went wrong.' ) . '</h1>' .
-		'<p>' . __( 'Invalid item ID.' ) . '</p>',
+		'<h1>' . __( 'An error occurred during the upload process.' ) . '</h1>' .
+		'<p>' . __( 'Invalid item ID. Please try again.' ) . '</p>',
```

Comment:
Trying again with an invalid ID will continue to fail.

Maybe add a link to the media library and direct the user there.

```suggestion
		'<p>' . __( 'Invalid item ID.' ) . '</p>',
```